.github/workflows/weekly.yml

```
@@ -0,0 +1,142 @@
+name: Weekly beta compile test
+
+on:
+  schedule:
+    # New versions of rust release on Thursdays. We test on Wednesdays to get at least a day's warning before all our CI breaks again.
+    # https://forge.rust-lang.org/release/process.html#release-day-thursday
+    - cron:  '0 12 * * 3'
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+  NIGHTLY_TOOLCHAIN: nightly
+
+jobs:
+  build:
```

If this is running the test suite, should it be renamed to `test`?

```suggestion
  test:
```